transport/rif_comms/client_exception_handler.py

```diff
@@ -0,0 +1,21 @@
+from typing import Union
+
+from grpc import RpcError, StatusCode
+
+from transport.rif_comms.exceptions import ClientException, TimeoutException, InternalException, \
+    InvalidArgumentException, NotFoundException, FailedPreconditionException
+
+
+class ClientExceptionHandler:
+    EXCEPTION_MAPPING = {
+        StatusCode.DEADLINE_EXCEEDED: TimeoutException,
+        StatusCode.INTERNAL: InternalException,
+        StatusCode.INVALID_ARGUMENT: InvalidArgumentException,
+        StatusCode.NOT_FOUND: NotFoundException,
+        StatusCode.FAILED_PRECONDITION: FailedPreconditionException
+    }
+
+    @classmethod
+    def get_exception(cls, rpc_error: RpcError) -> Union[RpcError, ClientException]:
```

Comment:
i would also rename this to `map_exception`, `convert_exception`, `translate_exception`, etc.

Reply:
Basically this is get an Exception from an error, so translate doenst look good

map_exception, error_to_exception, convert_error looks better

get_exception(rpc error) seems good to me anyway, the purpose is get an exception instance from an error

Reply:
> Basically this is get an Exception from an error, so translate doenst look good
> 
> map_exception, error_to_exception, convert_error looks better
> 
> get_exception(rpc error) seems good to me anyway, the purpose is get an exception instance from an error

"get something" is really _so_ undescriptive to me. get it from where? from what, and how? i'm not a fan of the word `get` in almost any method name, to be honest.

even `error_to_exception` would have been better, but it's not a blocker anyways so it's fine :+1: